frontend/pages/queries/edit/EditQueryPage.tsx

```diff
@@ -319,7 +323,13 @@ const EditQueryPage = ({
 
   // Function instead of constant eliminates race condition
   const backToQueriesPath = () => {
-    return queryId ? PATHS.QUERY_DETAILS(queryId) : PATHS.MANAGE_QUERIES;
+    const manageQueryPage =
+      filteredQueriesPath ||
+      `${PATHS.MANAGE_QUERIES}?${buildQueryStringFromParams({
+        team_id: currentTeamId,
+      })}`;
+
+    return queryId ? PATHS.QUERY_DETAILS(queryId) : manageQueryPage;
```

Comment:
Switching to a team, going to run a live query, go back to any page, FIX still preserves team